src/Type.hs

```diff
@@ -73,23 +80,39 @@ newtype Ty = Ty
   deriving (Eq, Show)
 makePrisms ''Ty
 
+tSyntax :: Ty
```

Comment:
As I mentioned elsewhere, this could perhaps be easier to use by creating a type class and a couple of bidirectional pattern synonyms, right? Then users could basically ignore the newtype constructor.